marconi-sidechain/src/Marconi/Sidechain/Run.hs

```diff
@@ -35,10 +38,12 @@ run = do
 
     cliArgs@CliArgs{socketFilePath, networkId, optionsRetryConfig} <- parseCli
 
+    logInfo trace . Text.toStrict $ pShowDarkBg cliArgs
+
     securityParam <- withNodeConnectRetry trace optionsRetryConfig socketFilePath $ do
       Utils.toException $ Utils.querySecurityParam @Void networkId socketFilePath
 
-    rpcEnv <- mkSidechainEnvFromCliArgs securityParam cliArgs
+    rpcEnv <- mkSidechainEnvFromCliArgs securityParam cliArgs trace
     race_
-      (runHttpServer rpcEnv) -- Start HTTP server
-      (runSidechainIndexers trace cliArgs rpcEnv) -- Start the Sidechain indexers
+      (runReaderT runHttpServer rpcEnv) -- Start HTTP server
+      (runReaderT runSidechainIndexers rpcEnv) -- Start the Sidechain indexers
```

Comment:
I wonder if we shouldn't go with `ReaderT` in `runHttpServer` and `runSidechainIndexers` too (all the way down to where we're really using the reader context), but we could keep it for later refactoring.

Reply:
I agree. Let's do it now as we're already touching this part.

Reply:
As we discussed on the call, the changes for this could be somewhat complex. I'd like to get this ticket closed out, since it has already grown in scope a couple of times, so let's create a new ticket for continuing the `ReaderT` refactoring and I'll start working on it immediately.

Reply:
https://input-output.atlassian.net/browse/PLT-7446